### PR TITLE
#132: Create summary header section on Payments page.

### PR DESCRIPTION
**Pull request by 2Fast2BCn**

### CODE REVIEW

File: internal/summary/handlers.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	r, err := getPaymentsSummary(db, filter)
+	switch err.(type) {
+	case nil:
+		break
+	default:
+		server_errors.LogAndSendServerError(c, err)
+		return
+	}
```
Comment (maxwedwards) [inline internal/summary/handlers.go:48]: Simple `if err != nil` should work here right?

If you wanted to check the actual type of error, use `errors.Is` (checks value of error) and `errors.As` (checks type of error).

File: internal/summary/payments.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			`).
+		PlaceholderFormat(sq.Dollar).
+		From("payment").
+		Where(filter).
+		Prefix(`WITH
+			tz AS (select preferred_timezone as tz from settings),
+			pub_keys as (select array_agg(pub_key) from local_node)
+		`)
+
+	qs, args, err := qb.ToSql()
```
Comment (maxwedwards) [inline internal/summary/payments.go:31]: Just use a multiline golang string raw quote? It's what we have tended to do.

File: internal/summary/payments.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		&r.TotalAmt,
+		&r.TotalFees,
+	)
+
+	switch err {
+	case nil:
+		break
+	default:
+		return nil, err
+	}
```
Comment (maxwedwards) [inline internal/summary/payments.go:48]: Same as above, `if err != nil` should suffice

File: web/src/apiSlice.ts [hunk 4]
```diff
@@ -68,12 +68,18 @@ export const torqApi = createApi({
...
-    >({
+      >({
       query: ({ limit, offset, order, filter }) =>
         `payments?limit=${limit || 100}&offset=${offset || 0}${order ? "&order=" + JSON.stringify(order) : ""}${
           filter ? "&filter=" + JSON.stringify(filter) : ""
         }`,
     }),
+    getPaymentsSummary: builder.query<
+      any,
+      { filter?: any }
```
Comment (maxwedwards) [inline web/src/apiSlice.ts:79]: We are trying to move away from using `any`s and always using specific types or `unknown` if it is truly unknown. In this case I think we know the type.

File: web/src/features/templates/tablePageTemplate/table-page-template.module.scss [hunk 5]
```diff
@@ -50,6 +50,57 @@
   grid-area: Totals;
   display: grid;
 }
+
+.pageRow {
+  display: grid;
```
Comment (maxwedwards) [inline web/src/features/templates/tablePageTemplate/table-page-template.module.scss:55]: Due to some performance problems in safari we have been moving to using flexbox over grid. I'm not sure it matters much here but seems to matter a lot more when there are a lot of them such as in a table of data.

Thoughts @ohenrik ?

File: web/src/features/transact/Payments/PaymentsHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import styles from "../../templates/tablePageTemplate/table-page-template.module.scss";
```
Comment (maxwedwards) [inline web/src/features/transact/Payments/PaymentsHeader.tsx:1]: Are these styles really going to be used anywhere else or should they be in a module.scss file local and specific to this file?

File: web/src/features/transact/Payments/PaymentsHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import classNames from "classnames";
+import * as d3 from "d3";
+import {useGetPaymentsSummaryQuery} from "../../../apiSlice";
+import {deserialiseQuery} from "../../sidebar/sections/filter/filter";
+import {useAppSelector} from "../../../store/hooks";
```
Comment (maxwedwards) [inline web/src/features/transact/Payments/PaymentsHeader.tsx:6]: The way we have setup our frontend build is that you don't need to specify relative paths like this. We told node to look in the `web/src` folder for packages. This means that `import {useGetPaymentsSummaryQuery} from "apiSlice";` would work.

File: web/src/features/transact/Payments/PaymentsPage.tsx [hunk 8]
```diff
@@ -269,10 +270,16 @@ function PaymentsPage() {
       offsetHandler={setOffset}
     />
   );
+
+  const pageTotals = (
```
Comment (maxwedwards) [inline web/src/features/transact/Payments/PaymentsPage.tsx:274]: uneeded variable? can just be inlined

File: web/src/features/transact/Payments/PaymentsPage.tsx [hunk 9]
```diff
@@ -269,10 +270,16 @@ function PaymentsPage() {
...
+    <PaymentsHeader />
+  );
+
   return (
     <TablePageTemplate
       title={"Payments"}
       breadcrumbs={breadcrumbs}
+      pageTotals={pageTotals}
```
Comment (maxwedwards) [inline web/src/features/transact/Payments/PaymentsPage.tsx:282]: as above, just inline?

File: cmd/torq/internal/torqsrv/server.go [hunk 10]
```diff
@@ -144,6 +145,11 @@ func registerRoutes(r *gin.Engine, db *sqlx.DB, apiPwd string, restartLNDSub fun
 			flow.RegisterFlowRoutes(flowRoutes, db)
 		}
 
+		summaryRoutes := api.Group("/summary")
+		{
+			summary.RegisterSummaryRoutes(summaryRoutes, db)
+		}
```
Comment (maxwedwards) [inline cmd/torq/internal/torqsrv/server.go:151]: Do we really need a new route for this? Can't the data be summarised on the front end from an existing api call with the full data?

Comment (2Fast2BCn) [reply cmd/torq/internal/torqsrv/server.go:151]: I wrote a separate query for performance reasons. I didn't look into the codebase yet on how paging is done but if it returns partial data with offsets then I'm not sure using an existing api call retrieving all data only to show a summary is the best option?

In my opinion the summary should be for all pages not only the visible one.

File: web/src/features/transact/Payments/PaymentsHeader.tsx (hunk 6 above)
Comment (2Fast2BCn) [reply web/src/features/transact/Payments/PaymentsHeader.tsx:1]: I think this style of table header might be added on other pages too right? If not then it makes sense to move it.

File: web/src/features/templates/tablePageTemplate/table-page-template.module.scss (hunk 5 above)
Comment (2Fast2BCn) [reply web/src/features/templates/tablePageTemplate/table-page-template.module.scss:55]: I kinda stole the layout from web/src/features/channel/channel-page.module.scss

**Comment by maxwedwards** [comment]
Going to close for now until this comes back with clearer figma design and requirements

**Pull request closed without merging**